modules/imap/functions.php

```diff
@@ -261,18 +261,21 @@ function format_imap_message_list($msg_list, $output_module, $parent_list=false,
         if ($is_snoozed) {
             $snooze_header = parse_delayed_header('X-Snoozed: '.$msg['x_snoozed'], 'X-Snoozed');
             $date = $snooze_header['until'];
-            $timestamp = strtotime($date);
+            $convertedDate = $date instanceof DateTime ? $date->format('Y-m-d H:i:s') : $date;
+            $timestamp = strtotime($convertedDate);
         } elseif ($is_scheduled) {
             $date = $msg['x_schedule'];
-            $timestamp = strtotime($date);
+            $convertedDate = $date instanceof DateTime ? $date->format('Y-m-d H:i:s') : $date;
+            $timestamp = strtotime($convertedDate);
         } else {
             if ($list_sort == 'date') {
                 $date_field = 'date';
             } else {
                 $date_field = 'internal_date';
             }
-            $date = translate_time_str(human_readable_interval($msg[$date_field]), $output_module);
-            $timestamp = strtotime($msg[$date_field]);
+            $convertedDate = $msg[$date_field] instanceof DateTime ? $msg[$date_field]->format('Y-m-d H:i:s') : $msg[$date_field];
```

Comment:
Please standardize the format of the dates returned by hm-mailbox no matter what the underlying connection is. I see it used to parse strings to dates, so this code expects strings. Leave it like this to expect strings and instead modify offending code in hm-ews to return strings of dates. Otherwise, you can refactor imap module to return DateTime objects but make sure hm-mailbox always returns the same type of data. I'd go with strings as ews is an add-on to imap and is more rarely used, so we should keep focus on imap mainly.